articles/openshift/howto-deploy-java-eap-app.md

```diff
@@ -0,0 +1,384 @@
+1. Install a Java SE implementation (for example, [AdoptOpenJDK OpenJDK 11 LTS/OpenJ9](https://adoptopenjdk.net/?variant=openjdk11&jvmVariant=openj9)).
+1. Install [Maven](https://maven.apache.org/download.cgi) 3.6.3 or higher.
+1. Install [Docker](https://docs.docker.com/get-docker/) for your OS.
+1. Install [Azure CLI](/cli/azure/install-azure-cli) 2.29.2 or later.
+1. Clone the code for this demo application on your local system. The demo application is on [GitHub](https://github.com/yersan/todo-list).
+1. Follow the instructions in [Create an Azure Red Hat OpenShift 4 cluster](./tutorial-create-cluster.md).
+
+   Though the "Get a Red Hat pull secret" step is labeled as optional, **it is required for this article**.  The pull secret enables your Azure Red Hat OpenShift cluster to find the JBoss EAP application images.
+
+   If you plan to run memory-intensive applications on the cluster, specify the proper virtual machine size for the worker nodes using the `--worker-vm-size` parameter. For more information, see:
+
+   * [Azure CLI to create a cluster](/cli/azure/aro#az_aro_create)
+   * [Supported virtual machine sizes for memory optimized](./support-policies-v4.md#memory-optimized)
+
+1. Connect to the cluster by following the steps in [Connect to an Azure Red Hat OpenShift 4 cluster](./tutorial-connect-cluster.md).
+   * Follow the steps in "Install the OpenShift CLI" 
+   * Connect to an Azure Red Hat OpenShift cluster using the OpenShift CLI with the user `kubeadmin`
+
+1. Execute the following command to create the OpenShift project for this demo application:
+
+    ```bash
+    $ oc new-project eap-demo
+    Now using project "eap-demo" on server "https://api.zhbq0jig.northeurope.aroapp.io:6443".
+
+    You can add applications to this project with the 'new-app' command. For example, try:
+
+    oc new-app rails-postgresql-example
+
+    to build a new example application in Ruby. Or use kubectl to deploy a simple Kubernetes application:
+
+    kubectl create deployment hello-node --image=k8s.gcr.io/serve_hostname
+    ```
+
+1. Execute the following command to add the view role to the default service account. This role is needed so the application can discover other pods and form a cluster with them:
+
+    ```bash
+    $ oc policy add-role-to-user view system:serviceaccount:$(oc project -q):default -n $(oc project -q)
+    clusterrole.rbac.authorization.k8s.io/view added: "system:serviceaccount:eap-demo:default"
+    ```
+
+## Prepare the application
+
+At this stage, you have cloned the demo application and your local repository is on the `main` branch. The demo application is a simple Jakarta EE 8 application that creates, reads, updates, and deletes records on a Microsoft SQL Server. Usually, you could have a JBoss EAP server installed on your local machine and configured with the required drivers and data source to allow connections from your application to the database server available on your environment.
```

Comment:
I am not sure what is intended by "Usually, you could have a a JBoss EAP server..."?

Maybe in context here and to satisfy some assumption it would be something like, "In order to test the application locally, it is also expected that you already have a JBoss EAP server..."

(also a bit later in the statement a mix of plural and non-plural is present "drivers" vs "data source" - I am guessing here that "driver" is more accurate in that you would only need one of these adding to the server?)

Reply:
Assuming it's necessary, perhaps some statement about expecting EAP XP to have been installed would be added around here?

Reply:
OK, I understand better now the intention, thank you for clarifying. Maybe even changing "Usually" to "Traditionally" or something gives the impression of historical events